Yosemite/Yosemite/Tools/Media/ImageSourceWriter.swift

```diff
@@ -1,30 +1,33 @@
-/// Writes an image to a URL from a CGImageSource, via CGImageDestination, particular to the needs of a `MediaImageExporter`.
+/// Writes an image to a URL from a CGImageSource, particular to the needs of a `MediaImageExporter`.
 ///
-struct ImageSourceWriter {
-
-    /// File URL where the image should be written.
-    ///
-    private let url: URL
-
-    /// The UTType of the image source.
+protocol ImageSourceWriter {
+    /// Write a given image source, succeeds unless an error is thrown, returns the resulting properties if available.
     ///
-    private let sourceUTType: CFString
-
-    init(url: URL, sourceUTType: CFString) {
-        self.url = url
-        self.sourceUTType = sourceUTType
-    }
+    /// - Parameters:
+    ///   - source: Image source data.
+    ///   - url: File URL where the image should be written.
+    ///   - sourceUTType: The UTType of the image source.
+    ///   - options: Image export options.
+    func writeImageSource(_ source: CGImageSource,
+                          to url: URL,
+                          sourceUTType: CFString,
+                          options: MediaImageExportOptions) throws -> ImageSourceWriteResultProperties
+}
 
-    /// Struct for returned result from writing an image, and any properties worth keeping track of.
-    ///
-    struct WriteResultProperties {
-        let width: CGFloat?
-        let height: CGFloat?
-    }
+/// Struct for returned result from writing an image, and any properties worth keeping track of.
+///
+struct ImageSourceWriteResultProperties {
+    let width: CGFloat?
+    let height: CGFloat?
```

Comment:
I'm definitely missing some context here. Why can `width` and `height` be `nil`? Based on `DefaultImageSourceWriter.writeImageSource()`, it seems impossible for them to be `nil` at all. 🤔

Reply:
`ImageSourceWriteResultProperties` is actually not used for our image upload (the struct is from WPiOS), so I'm going to remove this struct and its usage. We can always add this back when we need it with unit tests